app/views/admin/products_v3/_variant_row.html.haml

```diff
@@ -30,14 +31,14 @@
   = error_message_on variant, :price
 %td.col-on_hand.field.popout{'data-controller': "popout"}
   %button.popout__button{'data-popout-target': "button", 'aria-label': t('admin.products_page.columns.on_hand')}
-    = variant.on_demand ? t(:on_demand) : variant.on_hand
+    = variant.send(method_on_demand) ? t(:on_demand) : variant.send(method_on_hand)
```

Comment:
Using `send` is a sign that there's an issue with the code design. I think that it would be better to implement the reader on the model to read the value from the database if the desired value isn't set.

Reply:
I have added 2 methods in `variant_stock` to choose from which field to read. I have applied the same logic as in the view.